test/integration/granting_permissions_test.rb

```diff
@@ -28,4 +28,48 @@ class GrantingPermissionsTest < ActionDispatch::IntegrationTest
 
     assert_include @user.permissions_for(app), 'write'
   end
+
+  should "log changed permissions if permissions changed" do
+    app = create(:application, name: "MyApp", with_supported_permissions: ["write"])
+
+    starting_log_count = @user.event_logs.count
+
+    visit edit_user_path(@user)
+    select "write", from: "Permissions for MyApp"
+
+    click_button "Update User"
+
+    assert_equal (starting_log_count + 1), @user.event_logs.count
+
+    last_log = @user.event_logs.last
+
+    assert_equal({
+      added: {
+        "MyApp" => ["write"]
+      }
+    }, last_log.data)
+
+    visit edit_user_path(@user)
+    click_button "Update User"
+
+    @user.reload
+
+    assert_equal (starting_log_count + 1), @user.event_logs.count
+  end
+
+  should "allow the admin to provide a change note" do
+    app = create(:application, name: "MyApp", with_supported_permissions: ["write"])
+
+    visit edit_user_path(@user)
+    select "write", from: "Permissions for MyApp"
+
+    within ".new-note" do
+      fill_in "Add a change note (optional)", with: "Completed author training"
+    end
+
+    click_button "Update User"
+
+    last_log = @user.event_logs.last
+    assert_equal "Completed author training", last_log.trailing_message
+  end
```

Comment:
should we make this test more comprehensive by asserting for presence of event log data on the event logs page instead of asserting on `event_logs.last`?